appservice/src/deploy/localGitDeploy.ts

```diff
@@ -35,6 +36,8 @@ export async function localGitDeploy(client: SiteClient, fsPath: string): Promis
         // tslint:disable-next-line:no-floating-promises
         localGit.push(remote, 'HEAD:master');
     } catch (err) {
+        const parsedError: IParsedError = parseError(err);
+        parsedError.message = maskValue(parsedError.message, nonNullProp(publishCredentials, 'publishingPassword'));
```

Comment:
You're still not masking all errors if you do it here since the force push code could throw an error. I suggest something like this to ensure that all of localGitDeploy is covered:
```typescript
export async function localGitDeploy(client: SiteClient, fsPath: string): Promise<void> {
    const publishCredentials: User = await client.getWebAppPublishCredential();
    try {
        await localGitDeployCore(client, fsPath, publishCredentials);
    } catch (error) {
        // mask publishCredentials.password
    }
}

export async function localGitDeployCore(client: SiteClient, fsPath: string, publishCredentials: User): Promise<void> {
...
```

NOTE: This would look obviously look different if you took my other suggestion below that has `callWithMaskHandling`